### PR TITLE
Use UUID type for UUID data

### PR DESCRIPTION
**Pull request by gavinleeedwards**
# Motivation and Context
Use UUID data type for columns that store UUID data.

# What has changed
Amended some column types to UUID.

# Checklist
Reminder: Make you have run `build_groundzero_ddl.sh` and not manually edited the ground zero DDL.
* [x] Have run automated script, not made manual changes

Reminder: Make sure to version tag this after the PR is merged
* [* ] Updated patch number
* [ *] Updated version_tag in `ddl_version.sql`
* [* ] Updated current_version in `patch_database.py`

# How to test?
Run this alongside all of the other `use-UUID-type-for-UUID-data` branches in case-processor, case-api, action-scheduler, action-processor and action-worker. 
Fire all of those up and run the acceptance tests against them.  

# Links
- https://trello.com/c/r9qwqhaS

### CODE REVIEW

**Comment by gavinleeedwards** [comment]
[Use UUID types for UUID data (5)](https://trello.com/c/r9qwqhaS/1119-use-uuid-types-for-uuid-data-5)

**Comment by gavinleeedwards** [comment]
> You haven't checked the box to confirm you ran `build_groundzero_ddl.sh` and didn't edit the DDL by hand

Sorry, ticked the others but missed that one 👍

File: groundzero_ddl/ACTION-ddl_version.sql [hunk 1]
```diff
@@ -3,5 +3,5 @@ CREATE TABLE ddl_version.version (version_tag varchar(256) PRIMARY KEY, updated_
 
 -- Version and patch number for the current ground zero,
 -- NOTE: These must be updated every time the repo is tagged
-INSERT INTO ddl_version.patches (patch_number, applied_timestamp) VALUES (100, current_timestamp);
-INSERT INTO ddl_version.version (version_tag, updated_timestamp) VALUES ('v1.0.0', current_timestamp);
+INSERT INTO ddl_version.patches (patch_number, applied_timestamp) VALUES (200, current_timestamp);
+INSERT INTO ddl_version.version (version_tag, updated_timestamp) VALUES ('v5.1.0', current_timestamp);
```
Comment (NickGrantONS) [inline groundzero_ddl/ACTION-ddl_version.sql:7]: nope

Comment (NickGrantONS) [reply groundzero_ddl/ACTION-ddl_version.sql:7]: should be v1.1.0 or v2.0.0

File: groundzero_ddl/ddl_version.sql [hunk 2]
```diff
@@ -3,5 +3,5 @@ CREATE TABLE ddl_version.version (version_tag varchar(256) PRIMARY KEY, updated_
 
 -- Version and patch number for the current ground zero,
 -- NOTE: These must be updated every time the repo is tagged
-INSERT INTO ddl_version.patches (patch_number, applied_timestamp) VALUES (2400, current_timestamp);
-INSERT INTO ddl_version.version (version_tag, updated_timestamp) VALUES ('v4.1.0', current_timestamp);
+INSERT INTO ddl_version.patches (patch_number, applied_timestamp) VALUES (2500, current_timestamp);
+INSERT INTO ddl_version.version (version_tag, updated_timestamp) VALUES ('v5.1.0', current_timestamp);
```
Comment (NickGrantONS) [inline groundzero_ddl/ddl_version.sql:7]: should be v5.0.0

Comment (NickGrantONS) [reply groundzero_ddl/ddl_version.sql:7]: or 4.2.0

File: patch_database.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 PATCHES_DIRECTORY_ACTION = Path(__file__).parent.joinpath('patches/action')
 
 # current_version should match the version in the ddl_version.sql file
-current_version = 'v4.1.0'
+current_version = 'v4.2.0'
```
Comment (NickGrantONS) [inline patch_database.py:12]: this doesn't match what's in `ddl-version.sql`

File: patch_database.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
...
 
 # current_version_action should match the version in the ACTION-ddl_version.sql file
 current_version_action = 'v1.0.0'
```
Comment (NickGrantONS) [inline patch_database.py:15]: This doesn't match what's in `ACTION-ddl-version.sql`

File: groundzero_ddl/ddl_version.sql (hunk 2 above)
Comment (gavinleeedwards) [reply groundzero_ddl/ddl_version.sql:7]: A release 5.0.0 has already been cut so surely this one will have to be 5.1.0?

Comment (NickGrantONS) [reply groundzero_ddl/ddl_version.sql:7]: yep sorry missed that you're trying to match the tag to the version here

**Pull request merged**